docs/uninstallation_guide.md

```diff
@@ -17,4 +17,17 @@ Uninstall OSM [mesh name: osm] ? [y/n]: y
 OSM [mesh name: osm] uninstalled
 ```
 
-Run `osm mesh uninstall --help` for more options.
\ No newline at end of file
+Run `osm mesh uninstall --help` for more options.
+
+## Resource Management
+The following sections detail which Kubernetes resources are cleaned up and which remain after uninstalling OSM.
+### Removed during OSM uninstallation
+1. OSM controller resources (deployment, service, config map, and RBAC)
+1. Prometheus, Grafana, Jaeger, and Fluentbit resources installed by OSM
+1. Mutating webhook and validating webhook
+
+### Remaining after OSM uninstallation
+1. Existing Envoy sidecars
```

Comment:
Currently we create a k8s secret for each envoy that contains its bootstrap certificate. This is not being cleaned up as well.